File: shortestpathGraph.py
adjList = {
    1: [2, 5],
    2: [1, 3, 5],
    3: [2, 4],
    4: [3, 5],
    5: [1, 2, 4],
    6: []
}
def shortestPath(start,end):
    queue=[[start]]
    visited=set([start])
    print(queue)
    print(visited)
    while queue:
        curpath=queue.pop(0)
        curnode=curpath[len(curpath)-1]
        if curnode==end:
            return curpath
        for neighbour in adjList[curnode]:
            if neighbour not in visited:
                new=curpath+[neighbour]
                queue.append(new)
                visited.add(neighbour)
    return False
            

print(shortestPath(1,10))

File: shortestPathGraph.js
const edges=[
    ['w','x'],
    ['x','y'],
    ['z','y'],
    ['z','v'],
    ['w','v']
]
const shortestPath = (edges, nodeA,nodeB) =>{
    const graph = buildGraph(edges)
    let visited= new Set([nodeA])
    const queue = [[nodeA,0]]
    while(queue.length>0){
        const[node,distance]=queue.shift()
        if(node===nodeB) return distance
        for(let neighbour of graph[node]){
           if(!visited.has(neighbour)){
         queue.push([neighbour,distance+1])
         visited.add(neighbour)
           }    
        }
    }
    return -1
}
const buildGraph = (edges)=>{
    let graph ={}
    for (let edge of edges){
        const [a,b] =edge
        if(!(a in graph)) graph[a]=[]
        if(!(b in graph)) graph[b]=[]
        graph[a].push(b)
        graph[b].push(a)
    }
}  